uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -4253,6 +4223,188 @@ public function testTwoLeaveRequestsOneAMAndAnotherPMCanBeCreatedForTheSameDay()
+
+    $params['from_date'] = CRM_Utils_Date::processDate($date . ' 11:00:00');
+    $params['to_date'] = CRM_Utils_Date::processDate($date . ' 11:15:00');
+
+    $leaveRequest2 = LeaveRequestFabricator::fabricate($params, true);
+
+    $this->assertNotNull($leaveRequest1->id);
+    $this->assertNotNull($leaveRequest2->id);
+  }
+
+  public function testTwoLeaveRequestsBothForAMCannotBeCreatedForTheSameDay() {
+    $date = '2018-04-13';
+    $contactId = 1;
+    $absenceType = AbsenceTypeFabricator::fabricate();
+    $halfDayAMID = $this->leaveRequestDayTypes['half_day_am']['value'];
+
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2018-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2018-12-31'),
+    ]);
+
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $absenceType->id,
+      'contact_id' => $contactId,
+      'period_id' => $absencePeriod->id
+    ]);
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contactId],
+      ['period_start_date' => CRM_Utils_Date::processDate('2018-01-01')]
+    );
+
+    $this->createLeaveBalanceChange($periodEntitlement->id, 100);
+
+    $this->setExpectedException(
+      'CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException',
+      'This leave request overlaps with another request. Please modify dates of this request'
+    );
+
+    for ($i = 1; $i <= 2; $i++) {
+      LeaveRequestFabricator::fabricate([
+        'type_id' => $absenceType->id,
+        'contact_id' => $contactId,
+        'from_date' => CRM_Utils_Date::processDate($date),
+        'to_date' => CRM_Utils_Date::processDate($date),
+        'from_date_type' => $halfDayAMID,
+        'to_date_type' => $halfDayAMID
+      ], true);
+    }
+  }
+
+  public function testTwoLeaveRequestsBothForPMCannotBeCreatedForTheSameDay() {
+    $date = '2018-04-13';
+    $contactId = 1;
+    $absenceType = AbsenceTypeFabricator::fabricate();
+    $halfDayPMID = $this->leaveRequestDayTypes['half_day_pm']['value'];
+
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2018-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2018-12-31'),
+    ]);
+
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $absenceType->id,
+      'contact_id' => $contactId,
+      'period_id' => $absencePeriod->id
+    ]);
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contactId],
+      ['period_start_date' => CRM_Utils_Date::processDate('2018-01-01')]
+    );
+
+    $this->createLeaveBalanceChange($periodEntitlement->id, 100);
+
+    $this->setExpectedException(
+      'CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException',
+      'This leave request overlaps with another request. Please modify dates of this request'
+    );
+
+    for ($i = 1; $i <= 2; $i++) {
```

Comment:
Same as above, please store the parameters in a variable, and use it in the fabricator to create both requests. The advantage would be that we would know which of the requests threw the exception as the setExpectedException would be just before the second request.

Reply:
for var i = 1; i <= 2; i++; echo 'Done'.